accounting/accounting.go

```diff
@@ -302,7 +302,8 @@ func (accounting *State) AddTransaction(txnr *idb.TxnRow) (err error) {
 		accounting.updateAccountData(stxn.Txn.Sender, "spend", stxn.Txn.RekeyTo)
 	}
 
-	if stxn.Txn.Type == "pay" {
+	switch stxn.Txn.Type {
+	case atypes.PaymentTx:
```

Comment:
I'm using the accessors in the DB migration logic as well. If not for that I would be fine with the inline checks. What I'd really like is to find an abstraction that lets this entire accounting function be reused in the migration. Maybe it can be, after I get the regression tests written I can give that a try.

I do happen to think this change would make it easier for someone trying to learn from the code, even without the code reuse. Some of these checks would not be obvious to someone who isn't well versed in Algorand transactions. For example, App OptOut is `stxn.Txn.OnCompletion == sdk_types.CloseOutOC || stxn.Txn.OnCompletion == sdk_types.ClearStateOC`, I think having a name like `AppOptOutTxn` makes things clearer. Same with most of these, to be honest, even a simple expression like `stxn.Txn.AssetParams.IsZero()` isn't obviously an AssetDestroy.

Reply:
Sleeping on it I figured out what I wish we had was the accessors built into the SignedTxnInBlock type. They'd be more natural member functions there.

Reply:
I like this idea, I added one of those "TODO: Move to SDK" comments to make sure it doesn't get forgotten.